### PR TITLE
Allow to pass supported scala version url from command line

### PR DESCRIPTION
**Pull request by lwronski**

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala [hunk 1]
```diff
@@ -288,7 +293,8 @@ final case class SharedOptions(
             dependencies.compilerPlugin.map(Positioned.none),
             ignoreErrors
           ),
-        platform = platformOpt.map(o => Positioned(List(Position.CommandLine()), o))
+        platform = platformOpt.map(o => Positioned(List(Position.CommandLine()), o)),
+        supportedScalaVersionsUrl = supportedScalaVersionsUrl
```
Comment (kubukoz) [inline modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala:297]: **question:** is the output cached anywhere? I don't suppose we call this URL on every CLI command

**Comment by Gedochao** [comment]
Can we add some tests?

File: modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala (hunk 1 above)
Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala:297]: @kubukoz, I have another question about the root cause of your problem. Can you attach the error message that you encounter? Is it related to a proxy issue?

Comment (kubukoz) [reply modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala:297]: It's related to a issue of building inside Nix's sandbox - I need a reproducible way to download everything that needs to be downloaded. I'm planning to post a larger issue with things that would help Nix usage.

Unless other people need this solution, I would suggest not spending too much time until I describe the usecase in more context - maybe there's a better way to solve my problem :)

Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala:297]: Yes, I agree that providing more context would be much better. I will close this pull request as it isn't very useful for user.

**Pull request closed without merging**